src/lib/text-field/text-field-adapter.ts

```diff
@@ -1,130 +1,187 @@
+          const element = mutation.target as HTMLElement;
+          const attribute = element.getAttribute(mutation.attributeName);
+          const attributeName = mutation.attributeName as keyof typeof TEXT_FIELD_CONSTANTS.observedInputAttributes;
+          listener(attributeName, attribute);
+        }
+      });
+    });
+    this._inputElements.forEach(el => {
+      this._inputMutationObserver?.observe(el, { attributes: true, attributeFilter: TEXT_FIELD_CONSTANTS.observedInputAttributes });
+
+      // Call the listener with each observed attribute to capture the initial state
+      Object.values(TEXT_FIELD_CONSTANTS.observedInputAttributes).forEach(value => {
+        const attributeName = value as keyof typeof TEXT_FIELD_CONSTANTS.observedInputAttributes;
+        listener(attributeName, el.getAttribute(attributeName as string));
+      });
     });
   }
 
-  public override inputHasValue(): boolean {
-    return this._inputsSome(input => input.value ? input.value.trim().length > 0 : false);
-  }
+  public tryAddValueChangeListener(context: unknown, listener: TextFieldValueChangeListener): void {
+    // Destroy the previous value change listeners
+    this._destroyValueChangerListeners.forEach(callback => callback());
 
-  public override hasPlaceholder(): boolean {
-    return this._inputsSome(input => input.placeholder ? input.placeholder.trim().length > 0 : false);
+    // Add a new value change listener to each input
+    this._destroyValueChangerListeners = this._inputElements.map(el => {
+      return listenOwnProperty(context, el, 'value', listener);
+    });
   }
 
-  public override inputHasFocus(target?: EventTarget | null): boolean {
-    const activeElement = getActiveElement(this._component.ownerDocument);
-    return this._inputsSome(input => input === target || input === activeElement);
+  public removeValueChangeListener(): void {
+    this._destroyValueChangerListeners.forEach(callback => callback());
   }
 
-  public override setInputClass(className: string): void {
-    this._applyToInputs(input => input.classList.add(className));
-  }
+  public tryFloatLabel(force?: boolean): void {
+    // Force the label to float if specified
+    if (force !== undefined) {
+      this._fieldElement.floatLabel = force;
+      return;
+    }
 
-  public override removeInputClass(className: string): void {
-    this._applyToInputs(input => input.classList.remove(className));
+    // Float the label if no input has a value or a placeholder
+    this._fieldElement.floatLabel = this.hasValue || this.hasPlaceholder;
   }
 
-  public override setInputAttributeObserver(listener: (name: string, value: string) => void): void {
-    this._applyToInputs(input => {
-      const observer = createElementAttributeObserver(input, listener, FIELD_CONSTANTS.observedInputAttributes);
-      this._inputMutationObserverInstances.push(observer);
-    });
-  }
+  public tryConnectSlottedLabel(slot: HTMLSlotElement): void {
```

Comment:
When I was testing with a screen reader I noticed that the labels weren't getting attached to the `<input>` for both the native `<label>` and `<forge-label>`. I saw we have tests for this, but wasn't sure if I was missing something.

Reply:
Looking into it, the problem seems to be that when the example loads there is an `<input>` and a `<textarea>` slotted. Because there are two elements the component doesn't try to label either of them.